validator/nodejs/index.js

```diff
@@ -429,7 +540,7 @@ function main() {
     if (item === '-') {
       inputs.push(readFromStdin());
     } else if (isHttpOrHttpsUrl(item)) {
-      inputs.push(readFromUrl(item, program.userAgent));
+      inputs.push(crawlUrl(item, program));
```

Comment:
For each item in program.args, crawlUrl is started with a crawler - which then potentially adds more stuff to program.args. I wonder whether there's a race condition here, and what that means. Could it lead to an infinite crawl because each crawlUrl call has its own limit?
I understand this may not happen in practice, but I think it would be much better to avoid adding to program.args, and use something that makes it more explicit what kind of processing you intend to do with each input. E.g., a seperate list, or if it's some kind of fixed point iteration, some queue, etc. I would like to understand that the limit for the max crawl depth actually applies, at the moment I don't. Also to a lesser extent, it would be good to know that the number of crawl requests in flight are below the max concurrency as well (if there were multiple crawlers at the same time it wouldn't be true).